CMakeLists.txt

```diff
@@ -292,6 +292,7 @@ set(INCLUDED_DIRECTORIES
 	"dScripts/zone/PROPERTY/GF"
 	"dScripts/zone/PROPERTY/NS"
 
+	"thirdparty/magic_enum/include/magic_enum"
```

Comment:
if we are vendoring magic_enum, we should be using their cmake instead of just including their directories as we do for the other third parties